### PR TITLE
chore: automate prettier in docs

### PR DESCRIPTION
**Pull request by theDavidBarton**
Follow up to https://github.com/puppeteer/puppeteer/pull/7012#pullrequestreview-617175720

> mathiasbynens left a comment:
Nice! Should we automate this?

Why not? Here is my proposal.

**What have I done?**
- check and fix with prettier the following markdown files (that usually contains JavaScript code snippets): `**/README.md`, `docs/api.md`, `docs/troubleshooting.md` (change is in [`package.json`](https://github.com/puppeteer/puppeteer/pull/7014/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519))
- apply [`prettier-ignore`](https://prettier.io/docs/en/ignore.html#markdown) on auto-generated sections, and vendor docs
- mention the autofix option in [`CONTRIBUTING.md`](https://github.com/puppeteer/puppeteer/pull/7014/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055) > _Writing Documentation_
- applied the fixes on the current doc files
- [x] TODO: fix the failing formatting checks after the first, experimental commits

**Known regressions on the markdown files:**
- prettier replaced asterisk characters `*` with underscores `_` in case of italic texts
- prettier-ignore comments require empty lines before and after, it makes the raw markdown files a bit uglier
- newlines inserted after headings
- in general, this PR has a similar effect on the snippets as [#5825/files](https://github.com/puppeteer/puppeteer/pull/5825/files) had on the Puppeteer code base.

I hope you will like it (at least a little 😅 ).

_FYI:_ the first commit(s) will fail by purpose (see _TODO_).

### CODE REVIEW

**Comment by theDavidBarton** [comment]
Much more commits in the end than would be justified, but at least I've improved my stats for March. 😅

File: docs/api.md [hunk 1]
```diff
@@ -432,8 +444,7 @@ If Puppeteer doesn't find them in the environment during the installation step,
 - `PUPPETEER_EXECUTABLE_PATH` - specify an executable path to be used in `puppeteer.launch`. See [puppeteer.launch([options])](#puppeteerlaunchoptions) on how the executable path is inferred. **BEWARE**: Puppeteer is only [guaranteed to work](https://github.com/puppeteer/puppeteer/#q-why-doesnt-puppeteer-vxxx-work-with-chromium-vyyy) with the bundled Chromium, use at your own risk.
 - `PUPPETEER_PRODUCT` - specify which browser you'd like Puppeteer to use. Must be one of `chrome` or `firefox`. This can also be used during installation to fetch the recommended browser binary. Setting `product` programmatically in [puppeteer.launch([options])](#puppeteerlaunchoptions) supersedes this environment variable. The product is exposed in [`puppeteer.product`](#puppeteerproduct)
 
-> **NOTE** PUPPETEER_* env variables are not accounted for in the [`puppeteer-core`](https://www.npmjs.com/package/puppeteer-core) package.
-
+> **NOTE** PUPPETEER\_\* env variables are not accounted for in the [`puppeteer-core`](https://www.npmjs.com/package/puppeteer-core) package.
```
Comment (mathiasbynens) [inline docs/api.md]:
```suggestion
> **NOTE** `PUPPETEER_*` env variables are not accounted for in the [`puppeteer-core`](https://www.npmjs.com/package/puppeteer-core) package.
```

File: vendor/mitt/README.md [hunk 2]
```diff
@@ -177,3 +179,5 @@ Pull requests are the greatest contributions, so be sure they are focused in sco
 ## License
 
 [MIT License](https://opensource.org/licenses/MIT) © [Jason Miller](https://jasonformat.com/)
+
+<!-- prettier-ignore-end -->
```
Comment (mathiasbynens) [inline vendor/mitt/README.md]: Let's not change vendored subdirectories. Instead let's exclude `vendor/*` from the script (perhaps using [`.prettierignore`](https://prettier.io/docs/en/ignore.html#ignoring-files-prettierignore)?)

File: utils/doclint/README.md [hunk 3]
```diff
@@ -6,9 +6,9 @@ Puppeteer's source code.
 Doclint works in a few steps:
 
 1. Read sources in `lib/` folder, parse AST trees and extract public API
-    - note that we run DocLint on the outputted JavaScript in `lib/` rather than the source code in `src/`. We will do this until we have migrated `src/` to be exclusively TypeScript and then we can update DocLint to support TypeScript.
+   - note that we run DocLint on the outputted JavaScript in `lib/` rather than the source code in `src/`. We will do this until we have migrated `src/` to be exclusively TypeScript and then we can update DocLint to support TypeScript.
 2. Read sources in `docs/` folder, render markdown to HTML, use puppeteer to traverse the HTML
-  and extract described API
+   and extract described API
```
Comment (mathiasbynens) [inline utils/doclint/README.md]: GitHub won't let me make the suggestion, but this should be on the same line. Same for `1.` above.

**Review by mathiasbynens — CHANGES_REQUESTED** [review]

File: docs/api.md (hunk 1 above)
Comment (theDavidBarton) [reply docs/api.md]: Thanks, I've applied the change in https://github.com/puppeteer/puppeteer/pull/7014/commits/1f394174d366ff0654f9a4f699591785412a3f08.

File: vendor/mitt/README.md (hunk 2 above)
Comment (theDavidBarton) [reply vendor/mitt/README.md]: Thanks, indeed much nicer suggestion I've introduced the ignore file in https://github.com/puppeteer/puppeteer/pull/7014/commits/1f394174d366ff0654f9a4f699591785412a3f08.

File: utils/doclint/README.md (hunk 3 above)
Comment (theDavidBarton) [reply utils/doclint/README.md]: I've applied the change in 1f39417.

File: .prettierignore [hunk 4]
```diff
@@ -0,0 +1,8 @@
+node_modules/
+lib/
+third_party/*
```
Comment (mathiasbynens) [inline .prettierignore]:
```suggestion
third_party/
```

File: experimental/puppeteer-firefox/README.md [hunk 5]
```diff
@@ -21,7 +21,7 @@ Note: When you install puppeteer-firefox, it downloads a [custom-built Firefox](
 
 ### Usage
 
-**Example** - navigating to https://example.com and saving a screenshot as *example.png*:
+**Example** - navigating to https://example.com and saving a screenshot as _example.png_:
```
Comment (mathiasbynens) [inline experimental/puppeteer-firefox/README.md]:
```suggestion
**Example** - navigating to https://example.com and saving a screenshot as `example.png`:
```

**Review by mathiasbynens — APPROVED** [review]
Some final nits, then LGTM!

**Comment by mathiasbynens** [comment]
Thanks!

**Pull request merged**